### PR TITLE
Mob 3228

### PR DESCRIPTION
**Pull request by cosminBadulescu**
## JIRA
[MOB-3228](https://karhoo.atlassian.net/browse/MOB-3228)

## DESCRIPTION
[Create a description of what you have changed in this branch]

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [ ] New feature 
- [ ] Tech Debt
- [x] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [ ] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -1,174 +1,76 @@
...
 
     defaultConfig {
-        applicationId "com.karhoo.karhootraveller"
-        minSdkVersion 21
-        targetSdkVersion 29
-        versionCode 143
-        versionName "1.12.0"
-        testInstrumentationRunner "com.karhoo.karhootraveller.common.testrunner.TravellerTestRunner"
+        applicationId "com.karhoo.samples.uisdk.dropin"
+        minSdkVersion 22
```
Comment (jsm20) [inline app/build.gradle:12]: Has this been agreed? Assume it shouldn't be a problem but wasn't sure if this needs to be updated in any documentation?

**Comment by jsm20** [comment]
There seem to be issues with Karhoo user and auth token login for the UISDK sample app but working for guest users. Not sure if we want to fix these now or get these changes in first and then fix these issues?

File: app/src/main/java/com/karhoo/samples/uisdk/dropin/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                GuestConfig(
+                    applicationContext
+                )
+            )
+        }
+    }
+
+    private fun loginTokenExchange() {
+        KarhooApi.userService.logout()
+        val token: String = BuildConfig.BRAINTREE_AUTH_TOKEN
```
Comment (jsm20) [inline app/src/main/java/com/karhoo/samples/uisdk/dropin/MainActivity.kt:102]: Needs to be able to handle Adyen auth token as well.

File: app/build.gradle (hunk 1 above)
Comment (mohadian) [reply app/build.gradle:12]: I guess this is for the sample app. we are not documenting this one. The one for the UI SDK perhaps need to be documented

File: app/src/test/java/com/karhoo/samples/uisdk/dropin/ExampleUnitTest.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import org.junit.Test
+
+import org.junit.Assert.*
+
+/**
+ * Example local unit test, which will execute on the development machine (host).
+ *
+ * See [testing documentation](http://d.android.com/tools/testing).
+ */
+class ExampleUnitTest {
```
Comment (mohadian) [inline app/src/test/java/com/karhoo/samples/uisdk/dropin/ExampleUnitTest.kt:12]: We probably dont need it

File: app/src/main/res/values/themes.xml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+        <item name="colorPrimaryVariant">@color/purple_700</item>
+        <item name="colorOnPrimary">@color/white</item>
+        <!-- Secondary brand color. -->
+        <item name="colorSecondary">@color/teal_200</item>
+        <item name="colorSecondaryVariant">@color/teal_700</item>
+        <!-- Status bar color. -->
+        <item name="android:statusBarColor" tools:targetApi="l">?attr/colorPrimaryVariant</item>
+        <!-- Customize your theme here. -->
+    </style>
+</resources>
```
Comment (mohadian) [inline app/src/main/res/values/themes.xml:15]: I thought Detekt would complain about not having `NewLineAtEndOfFile`

File: app/src/main/res/values/colors.xml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <color name="purple_200">#FFBB86FC</color>
+    <color name="purple_500">#FF6200EE</color>
+    <color name="purple_700">#FF3700B3</color>
+    <color name="teal_200">#FF03DAC5</color>
+    <color name="teal_700">#FF018786</color>
+    <color name="black">#FF000000</color>
+    <color name="white">#FFFFFFFF</color>
+</resources>
```
Comment (mohadian) [inline app/src/main/res/values/colors.xml:10]: Same 👇  (Does not Detekt complain about NewLineAtEndOfFile)

File: app/src/androidTest/java/com/karhoo/samples/uisdk/dropin/MainRobot.kt [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+                resId = R.id.passwordInput,
+                text = "AsA!12345678Aa"
+        )
+    }
+
+    fun userClicksOnSignInButton() {
+        clickButton(R.id.signInButton)
+    }
+
+}
```
Comment (mohadian) [inline app/src/androidTest/java/com/karhoo/samples/uisdk/dropin/MainRobot.kt:54]: Does not Detek complain about NewLineAtEndOfFile?

File: app/src/androidTest/java/com/karhoo/samples/uisdk/dropin/ExampleInstrumentedTest.kt [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+import org.junit.Assert.*
+
+/**
+ * Instrumented test, which will execute on an Android device.
+ *
+ * See [testing documentation](http://d.android.com/tools/testing).
+ */
+@RunWith(AndroidJUnit4::class)
+class ExampleInstrumentedTest {
```
Comment (mohadian) [inline app/src/androidTest/java/com/karhoo/samples/uisdk/dropin/ExampleInstrumentedTest.kt:17]: we probably don't need this test

File: app/build.gradle (hunk 1 above)
Comment (jsm20) [reply app/build.gradle:12]: Do we want to keep the two in line wrt to versions supported?

File: app/src/main/java/com/karhoo/samples/uisdk/dropin/MainActivity.kt (hunk 2 above)
Comment (jsm20) [reply app/src/main/java/com/karhoo/samples/uisdk/dropin/MainActivity.kt:102]: Also the same for Guest User. Would be better to be able to switch easily between Adyen and Braintree.

File: app/src/test/java/com/karhoo/samples/uisdk/dropin/ExampleUnitTest.kt (hunk 3 above)
Comment (cosminBadulescu) [reply app/src/test/java/com/karhoo/samples/uisdk/dropin/ExampleUnitTest.kt:12]: removed

File: app/src/androidTest/java/com/karhoo/samples/uisdk/dropin/MainRobot.kt (hunk 6 above)
Comment (cosminBadulescu) [reply app/src/androidTest/java/com/karhoo/samples/uisdk/dropin/MainRobot.kt:54]: as i can see no, but not sure why... i will add it anyway

File: app/src/main/res/values/colors.xml (hunk 5 above)
Comment (cosminBadulescu) [reply app/src/main/res/values/colors.xml:10]: as i can see no, but not sure why... i will add it anyway

File: app/src/main/res/values/themes.xml (hunk 4 above)
Comment (cosminBadulescu) [reply app/src/main/res/values/themes.xml:15]: as i can see no, but not sure why... i will add it anyway

File: app/src/androidTest/java/com/karhoo/samples/uisdk/dropin/ExampleInstrumentedTest.kt (hunk 7 above)
Comment (cosminBadulescu) [reply app/src/androidTest/java/com/karhoo/samples/uisdk/dropin/ExampleInstrumentedTest.kt:17]: removed

File: app/build.gradle (hunk 1 above)
Comment (cosminBadulescu) [reply app/build.gradle:12]: i left it as it was in the Dropin sample app, but i will lower it to 21

**Comment by jsm20** [comment]
I think some tweaking still required for the sample app but mostly working so better to get this merged in first and those can be tackled in a separate PR.

**Review by jsm20 — APPROVED** [review]

**Pull request merged**